### DISCUSSION TITLE
Relative import path "react" not prefixed (Deno)

### DISCUSSION THREAD

**Discussion by JacobWeisenburger** [pmndrs/jotai#1877] [Q&A] [answered]
How do we get around this error? I would like to be able to use jotai in Deno with vanilla js. Is this possible?

```ts
import { atom } from 'https://deno.land/x/jotai@v2.0.4/src/index.ts'
```

```
error: Relative import path "react" not prefixed with / or ./ or ../ and not in import map from "https://deno.land/x/jotai@v2.0.4/src/react/Provider.ts"
    at https://deno.land/x/jotai@v2.0.4/src/react/Provider.ts:1:66
```

```
$ deno --version
deno 1.32.1 (release, x86_64-pc-windows-msvc)
v8 11.2.214.9
typescript 5.0.2
```

**Comment by dai-shi** [comment]
I think you need import maps for `react`.
If you don't use react, you might be able to import from jotai/vanilla.

Previous discussion: #1081

**Reply by JacobWeisenburger** [reply]
how do you read the value of an atom when using jotai/vanilla?

```ts
import { atom, useAtom } from 'https://deno.land/x/jotai@v2.0.4/src/vanilla/atom.ts'
// Module '"https://deno.land/x/jotai@v2.0.4/src/vanilla/atom.ts"' has no exported member 'useAtom'.
```

**Reply by dai-shi** [reply]
I haven't tried, but it should be something like this:
```ts
import { atom, createStore } from 'https://deno.land/x/jotai@v2.0.4/src/vanilla.ts';
import { useAtom } from 'https://deno.land/x/jotai@v2.0.4/src/react.ts';
```

useAtom requires React obvioiusly.

**Reply by JacobWeisenburger** [reply]
So it would appear that jotai can't be used with vanilla JS. Am I correct?

**Reply by dai-shi** [reply, marked as answer]
Depends on what you mean by "use". `jotai/vanilla` exposes vanilla API that you can use without React. It's not Deno specific. In Node, it will be something like this:

```js
import { atom, createStore } from 'jotai/vanilla';

const countAtom = atom(1);
const store = createStore();
store.set(countAtom, 1);
console.log(store.get(countAtom));
```

**Reply by JacobWeisenburger** [reply]
Thanks for your answer.

seems kind of convoluted that I would have to make a store just to read the value of an atom. I'm not sure this library is really ready for use outside of react. Back to Zustand I guess.

**Reply by dai-shi** [reply]
Yes, Zustand and Valtio are module state first and React binding afterwards.
Jotai is more React-ish. Usually, people don't need to use the store API, but in some edge cases it's useful.
We are still open for ideas about vanilla usage of Jotai, regardless of Node or Deno.

**Reply by JacobWeisenburger** [reply]
It seems to me that if you want jotai to be able to work without react, then it should have a way to get the value from an atom and perhaps a way to subscribe.

**Reply by dai-shi** [reply]
You have a misconception. An atom is just a definition like a function. It's not a store like Zustand.
Jotai store is like a WeakMap.
FYI: https://twitter.com/dai_shi/status/1631487223614046209
